android/app/src/main/java/com/microsoft/codepush/common/core/CodePushBaseCore.java

```diff
@@ -0,0 +1,1043 @@
+        mContext = context.getApplicationContext();
+        mIsDebugMode = isDebugMode;
+        if (serverUrl != null) {
+            mServerUrl = serverUrl;
+        }
+        mPublicKey = publicKeyProvider.getPublicKey();
+        mAppEntryPoint = appEntryPointProvider.getAppEntryPoint();
+        try {
+            PackageInfo pInfo = mContext.getPackageManager().getPackageInfo(mContext.getPackageName(), 0);
+            mAppVersion = pInfo.versionName;
+        } catch (PackageManager.NameNotFoundException e) {
+            throw new CodePushInitializeException("Unable to get package info for " + mContext.getPackageName(), e);
+        }
+
+        /* Initialize utilities. */
+        FileUtils fileUtils = FileUtils.getInstance();
+        CodePushUtils utils = CodePushUtils.getInstance(fileUtils);
+        CodePushUpdateUtils updateUtils = CodePushUpdateUtils.getInstance(fileUtils, utils);
+        mUtilities = new CodePushUtilities(utils, fileUtils, updateUtils, platformUtils);
+
+        /* Initialize managers. */
+        String documentsDirectory = mContext.getFilesDir().getAbsolutePath();
+        CodePushUpdateManager updateManager = new CodePushUpdateManager(documentsDirectory, platformUtils, fileUtils, utils, updateUtils);
+        SettingsManager settingsManager = new SettingsManager(mContext, utils);
+        CodePushTelemetryManager telemetryManager = new CodePushTelemetryManager(settingsManager);
+        CodePushRestartManager restartManager = new CodePushRestartManager(restartListener);
+        CodePushAcquisitionManager acquisitionManager = new CodePushAcquisitionManager(utils, fileUtils);
+        mManagers = new CodePushManagers(updateManager, telemetryManager, settingsManager, restartManager, acquisitionManager);
+
+        /* Initialize confirmation dialog for update install */
+        mConfirmationDialog = confirmationDialog;
+
+        /* Set current instance. */
+        mCurrentInstance = this;
+
+        /* Initialize state */
+        mState = new CodePushState();
+
+        /* Clear debug cache if needed. */
+        if (mIsDebugMode && mManagers.mSettingsManager.isPendingUpdate(null)) {
+            try {
+                mUtilities.mPlatformUtils.clearDebugCache();
+            } catch (IOException e) {
+                throw new CodePushInitializeException(e);
+            }
+        }
+
+        /* Initialize update after restart. */
+        try {
+            initializeUpdateAfterRestart();
+        } catch (CodePushGetPackageException | CodePushPlatformUtilsException | CodePushRollbackException e) {
+            throw new CodePushInitializeException(e);
+        }
+    }
+
+    /**
+     * Gets current app version.
+     *
+     * @return current app version.
+     */
+    public String getAppVersion() {
+        return mAppVersion;
+    }
+
+    /**
+     * Sets current app version.
+     *
+     * @param mAppVersion current app version.
+     */
+    public void setAppVersion(String mAppVersion) {
```

Comment:
Rename parameter in method to `appVersion`. "m" is used only for private properties, not parameters.

Reply:
@annakocheshkova I know, just missed it after refactoring. Thx anyway,